app_dart/lib/src/request_handlers/reset_prod_task.dart

```diff
@@ -120,49 +119,8 @@ class ResetProdTask extends ApiRequestHandler<Body> {
       return datastore.lookupByValue<Task>(key);
     }
 
-    final Key<String> commitKey = await _constructCommitKey(
-      datastore: datastore,
-      gitBranch: gitBranch!,
-      sha: sha!,
-      slug: slug!,
-    );
-
-    final Query<Task> query = datastore.db.query<Task>(ancestorKey: commitKey);
-    final List<Task> initialTasks = await query.run().toList();
-    log.fine('Found ${initialTasks.length} tasks for commit');
-    final List<Task> tasks = <Task>[];
-    log.fine('Searching for task with name=$name');
-    for (Task task in initialTasks) {
-      if (task.name == name) {
-        tasks.add(task);
-      }
-    }
-
-    if (tasks.length != 1) {
-      log.severe('Found ${tasks.length} entries for builder $name');
-      throw InternalServerError('Expected to find 1 task for $name, but found ${tasks.length}');
-    }
-
-    return tasks.single;
-  }
-
-  /// Construct the Datastore key for [Commit] that is the ancestor to this [Task].
-  ///
-  /// Throws [BadRequestException] if the given git branch does not exist in [CocoonConfig].
-  Future<Key<String>> _constructCommitKey({
-    required DatastoreService datastore,
-    required String gitBranch,
-    required String sha,
-    required RepositorySlug slug,
-  }) async {
-    gitBranch = gitBranch.trim();
-    sha = sha.trim();
-    final String id = '${slug.fullName}/$gitBranch/$sha';
-    final Key<String> commitKey = datastore.db.emptyKey.append<String>(Commit, id: id);
-    log.fine('Constructed commit key=$id');
-    // Return the official key from Datastore for task lookups.
-    final Commit commit = await datastore.lookupByValue<Commit>(commitKey);
-    return commit.key;
+    final Commit commit = await datastore.findCommit(gitBranch: gitBranch!, sha: sha!, slug: slug!);
+    return await datastore.findTask(commitKey: commit.key, name: name!);
```

Comment:
```suggestion
    return datastore.findTask(commitKey: commit.key, name: name!);
```

https://dart.dev/guides/language/effective-dart/usage#dont-use-async-when-it-has-no-useful-effect